src/tests/ftest/unittest/Unittest.py

```diff
@@ -25,65 +25,79 @@
 from general_utils import get_file_path
-from apricot import Test
+from apricot import Test, TestWithServers
 
-class UnitTest(Test):
+def unittest_runner(self, unit_testname):
     """
-    Avocado Unit Test class.
+    Common unitetest runner function.
+    Args:
+        unit_testname: unittest name.
+    return:
+        None
+    """
+    name = self.params.get("testname", '/run/UnitTest/{0}/'
+                           .format(unit_testname))
+    bin_path = get_file_path(name, "install/bin")
+
+    cmd = ("{0}".format(bin_path[0]))
+    return_code = process.system(cmd)
+    if return_code is not 0:
+        self.fail("{0} unittest failed with return code={1}.\n"
+                  .format(unit_testname, return_code))
+
+class UnitTestWithoutServers(Test):
+    """
+    Avocado Unit Test class for tests which don't need servers.
     :avocado: recursive
     """
     def tearDown(self):
         process.system("rm -f /mnt/daos/*")
 
-    def unittest_runner(self, unit_testname):
-        """
-        Common unitetest runner function.
-        Args:
-            unit_testname: unittest name.
-        return:
-            None
-        """
-        name = self.params.get("testname", '/run/UnitTest/{0}/'
-                               .format(unit_testname))
-        bin_path = get_file_path(name, "install/bin")
-
-        cmd = ("{0}".format(bin_path[0]))
-        return_code = process.system(cmd)
-        if return_code is not 0:
-            self.fail("{0} unittest failed with return code={1}.\n"
-                      .format(unit_testname, return_code))
-
     def test_smd_ut(self):
         """
         Test smd unittest.
         :avocado: tags=unittest,nvme,smd_ut
         """
-        self.unittest_runner("smd_ut")
+        unittest_runner(self, "smd_ut")
 
     def test_vea_ut(self):
         """
         Test vea unittest.
         :avocado: tags=unittest,nvme,vea_ut
         """
-        self.unittest_runner("vea_ut")
+        unittest_runner(self, "vea_ut")
 
     def test_pl_map(self):
         """
         Test pl_map unittest.
         :avocado: tags=unittest,pl_map
         """
-        self.unittest_runner("pl_map")
+        unittest_runner(self, "pl_map")
 
     def test_eq_tests(self):
         """
         Test eq_tests unittest.
         :avocado: tags=unittest,eq_tests
         """
-        self.unittest_runner("eq_tests")
+        unittest_runner(self, "eq_tests")
 
     def test_vos_tests(self):
         """
         Test eq_tests unittest.
         :avocado: tags=unittest,vos_tests
         """
-        self.unittest_runner("vos_tests")
+        unittest_runner(self, "vos_tests")
+
+class UnittestWithServers(TestWithServers):
+    """
+    Avocado Unit Test class for tests which need servers.
+    :avocado: recursive
+    """
+
+    def test_daos_addons(self):
+        """
+        Jira ID: DAOS-2326
+        Test daos_addons unittest.
+        :avocado: tags=unittest,daos_addons_test
```

Comment:
Should all of these tests have an `all` tag, at least one of the `smoke`, `pr`, or `full_regression` tags, a `small` tag in addition to what is already listed?  Allso the shouldn't the tests with the `nvme` tag also have the `hw` tag?  This would apply to the UnitTestWithoutServers test case tags as well.

Reply:
Done for this new added test. For rest of the tests, this has been taken care in patch for DAOS-2473. https://github.com/daos-stack/daos/pull/574/files#diff-590a5d80addb2f3118083e12c79a097b